### PR TITLE
Add a share manager that connects to oc10 databases

### PR DESCRIPTION
**Pull request by aduffeck**

### CODE REVIEW

File: pkg/share/manager/sql/conversions.go [hunk 1]
```diff
@@ -0,0 +1,259 @@
+// Copyright 2018-2021 CERN
```
Comment (ishank011) [inline pkg/share/manager/sql/conversions.go:1]: Maybe we can refactor `pkg/cbox/utils/conversions` to reduce duplicity?

File: pkg/share/manager/sql/sql.go [hunk 2]
```diff
@@ -0,0 +1,502 @@
...
+	"github.com/cs3org/reva/pkg/utils"
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+
+	// Provides mysql drivers
+	_ "github.com/go-sql-driver/mysql"
+)
+
+func init() {
+	registry.Register("sql", NewMysql)
```
Comment (ishank011) [inline pkg/share/manager/sql/sql.go:46]: @aduffeck can you register it as `oc10-sql` maybe? We already have a share driver with this name and there'll be a race condition which would affect which of the two drivers is served

Comment (aduffeck) [reply pkg/share/manager/sql/sql.go]: done

File: pkg/share/manager/sql/conversions.go (hunk 1 above)
Comment (aduffeck) [reply pkg/share/manager/sql/conversions.go:1]: The differences are big enough that my feeling is it's easier to keep them separate to be honest. Let me know if you want to go for it anyway and I'll see what I can do.

**Review by butonic — APPROVED** [review]

**Review by labkode — APPROVED** [review]

**Pull request merged**